java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/ImplementOverrideMethodGenerator.java

```diff
@@ -0,0 +1,178 @@
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+public final class ImplementOverrideMethodGenerator extends CodeGenerator {
+
+    public static final String GENERATE_IMPLEMENT_METHOD =  "java.generate.implementMethod";
+    public static final String GENERATE_OVERRIDE_METHOD =  "java.generate.overrideMethod";
+
+    private final Set<String> commands = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(GENERATE_IMPLEMENT_METHOD, GENERATE_OVERRIDE_METHOD)));
+    private final Gson gson = new Gson();
+
+    ImplementOverrideMethodGenerator() {
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "DN_GenerateImplementMethod=Generate Implement Method...",
+        "DN_GenerateOverrideMethod=Generate Override Method...",
+        "DN_From=(from {0})",
+    })
+    public List<CodeAction> getCodeActions(CompilationInfo info, CodeActionParams params) {
+        List<String> only = params.getContext().getOnly();
+        if (only == null || !only.contains(CodeActionKind.Source)) {
+            return Collections.emptyList();
+        }
+        int offset = getOffset(info, params.getRange().getStart());
+        TreePath tp = info.getTreeUtilities().pathFor(offset);
+        tp = info.getTreeUtilities().getPathElementOfKind(TreeUtilities.CLASS_TREE_KINDS, tp);
+        if (tp == null) {
+            return Collections.emptyList();
+        }
+        TypeElement typeElement = (TypeElement) info.getTrees().getElement(tp);
+        if (typeElement == null || typeElement.getKind() == ElementKind.ANNOTATION_TYPE) {
+            return Collections.emptyList();
+        }
+        List<CodeAction> result = new ArrayList<>();
+        String uri = Utils.toUri(info.getFileObject());
+        ElementUtilities eu = info.getElementUtilities();
+        if (typeElement.getKind().isClass() || typeElement.getKind().isInterface() && SourceVersion.RELEASE_8.compareTo(info.getSourceVersion()) <= 0) {
+            List<QuickPickItem> implementMethods = new ArrayList<>();
+            for (ExecutableElement method : eu.findUnimplementedMethods(typeElement, true)) {
+                boolean mustImplement = !method.getModifiers().contains(Modifier.DEFAULT);
+                Element enclosingElement = method.getEnclosingElement();
+                String enclosingTypeName = enclosingElement.getKind().isClass() || enclosingElement.getKind().isInterface() ? Bundle.DN_From(((TypeElement)enclosingElement).getQualifiedName().toString()) : null;
+                implementMethods.add(new QuickPickItem(createLabel(info, method), enclosingTypeName, null, mustImplement, new ElementData(method)));
+            }
+            if (!implementMethods.isEmpty()) {
+                result.add(createCodeAction(Bundle.DN_GenerateImplementMethod(), CODE_GENERATOR_KIND, GENERATE_IMPLEMENT_METHOD, uri, offset, implementMethods));
+            }
+        }
+        if (typeElement.getKind().isClass() || typeElement.getKind().isInterface()) {
```

Comment:
I am not sure what `GenreratorUtils.generateMethodOverrides` does when overriding method in the interface type for Java < 8, where `default` modified is not allowed.

Reply:
CodeAction is not available in interfaces for Java < 8 (see `getCodeActions(...)` method).